tests/integration_tests/Test/PM/Integration/Update.cls

```diff
@@ -0,0 +1,263 @@
+/// @params fromVersion         Version of module to install
+/// @params toVersion           Version of module to which to update
+/// @params developerMode       1 if update should be run with -dev flag, 0 otherwise
+Method ModuleNormalUpdateNoMirrorHelper(
+	moduleName As %String,
+	updateStepsToSeed As %DynamicArray,
+	updateStepsToRun As %DynamicArray,
+	fromVersion As %String,
+	toVersion As %String,
+	developerMode As %Boolean = 0)
+{
+  if '$system.Mirror.IsMember() {
+    // Load module
+    do ..AssertModuleInstalledCorrectly(moduleName, fromVersion)
+
+    // Verify that update steps were correctly seeded and capture the TimeStampStart property for comparison
+    set timeStampArr = []
+    set iter = updateStepsToSeed.%GetIterator()
+    while iter.%GetNext(.key, .value) {
+      set className = value.%Get(0)
+      set methodName = value.%Get(1)
+      set isPrimaryOnly = value.%Get(2)
+
+      set updateStep = ..GetUpdateStep(className, methodName, isPrimaryOnly)
+      do $$$AssertNotEquals("", updateStep, "Update step with class="_className_" and method="_methodName_" exists.")
+      set timeStampArr(updateStep.UpdateStepClassMethodIdx) = updateStep.TimeStampStart
+    }
+
+    // Update module
+    set devFlag = $select(developerMode: "-dev", 1: "")
+    set tSC = ##class(%IPM.Main).Shell("update -v "_devFlag_" "_moduleName_" "_toVersion)
+    do $$$AssertStatusOK(tSC, "Updated module "_moduleName_" without error")
+
+    // Check that seeded methods were not re-run
+    set iter = updateStepsToSeed.%GetIterator()
+    while iter.%GetNext(.key, .value) {
+      set className = value.%Get(0)
+      set methodName = value.%Get(1)
+      set isPrimaryOnly = value.%Get(2)
+
+      set updateStep = ..GetUpdateStep(className, methodName, isPrimaryOnly)
+      do $$$AssertEquals(timeStampArr(updateStep.UpdateStepClassMethodIdx), updateStep.TimeStampStart, "Seeded update step with class="_className_" and method="_methodName_" was correctly not rerun on update.")
+    }
+
+    // Check whether correct UpdateStep rows exist in the correct table after update command is run
+    do ..AssertUpdateStepsExist(updateStepsToRun)
+
+    // NOTE: For testing simplicity, we will always assume that in the update test, both the module and its dependencies are being updated
+    // from and to the same version numbers.
+
+    // Check module version has been updated
+    set module = ##class(%IPM.Storage.Module).NameOpen(moduleName)
+    do $$$AssertEquals(module.VersionString, toVersion)
+
+    // Check that dependency versions have been updated
+    // dependencyGraph format: dependencyGraph(<module name>) = $ListBuild(<depth>, <server>, <version>)
+    set moduleName = ""
+    for {
+      set moduleName = $order(dependencyGraph(moduleName),1,moduleInfo)
+      quit:moduleName=""
+
+      set moduleVersion = $list(moduleInfo, 2)
```

Comment:
@isc-eneil @isc-jili @isc-jlechtne  Looking at this made me realize one thing we do need to do (separate subtask to do later as an improvement): when updating a module, we should check all of its dependencies pre and post update and if there were any in pre that aren't in post, they should be uninstalled.